### ISSUE TITLE
Models which contain root elements lead to parser error

### ISSUE THREAD

**Issue by garyfeltham** [citypay/citypay-api-client-ruby#4]
The ruby parser expects data to be return in the format 
`{ "attribute": "value", "attribute2": "value2"...`

however the server returns a root element with each packet of data i.e.

`{ "className": { "attribute": "value", "attribute2": "value2"...`

This causes a Nil object to be instantiated.

**Comment by garyfeltham** [comment]
A fix is to ensure base line api models have the definition of an optional self check which allows for an object level conversion or a nested name level conversion for example

``` 
def self.build_from_hash(attributes)
        new.build_from_hash(attributes)
end```

becomes

```
def self.build_from_hash(attributes)
      if !attributes[:ClassName].nil?
        new.build_from_hash(attributes[:ClassName])
      else
        new.build_from_hash(attributes)
      end
end```

**Comment by garyfeltham** [comment]
Required unit testing confirmation

- Decision
- AuthResponse
- Acknowledgement
- AuthReferences
- ListMerchantsResponse
- CardHolderAccount

**Comment by garyfeltham** [comment]
Closed on 1.0.3
